apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java

```diff
@@ -759,6 +790,30 @@ static Reference createIdentifierReference(String identifierSystem, String ident
         .setDisplay(retrieveNpiCodeDisplay(identifierValue));
   }
 
+  /**
+   * Setting identifier reference when npiorgdisplay is passed in.
+   *
+   * @param identifierSystem the {@link Identifier#getSystem()} to use in {@link
+   *     Reference#getIdentifier()}
+   * @param identifierValue the {@link Identifier#getValue()} to use in {@link
+   *     Reference#getIdentifier()}
+   * @param npiOrgDisplay the {@link Identifier#getValue()} to use in {@link
+   *     Reference#getIdentifier()}
+   * @return a {@link Reference} with the specified {@link Identifier}
+   */
+  static Reference createIdentifierReference(
+      String identifierSystem, String identifierValue, Optional<String> npiOrgDisplay) {
+
+    if (!npiOrgDisplay.isEmpty()) {
+      return new Reference()
+          .setIdentifier(new Identifier().setSystem(identifierSystem).setValue(identifierValue))
+          .setDisplay(npiOrgDisplay.get());
+    } else {
+      return new Reference()
+          .setIdentifier(new Identifier().setSystem(identifierSystem).setValue(identifierValue));
+    }
+  }
```

Comment:
Since we're using optionals, we could simplify the code a little by using orElse like above:
```
return new Reference()
          .setIdentifier(new Identifier().setSystem(identifierSystem).setValue(identifierValue))
          .setDisplay(npiOrgDisplay.orElse(null));
```
Making some assumptions that display would be null if we didnt set it at all.

Reply:
Need to confirm

Reply:
confirmed